### ISSUE TITLE
Expose the wallet seed in the GUI for wallet backup/recovery

### ISSUE THREAD

**Issue by ITwrx** [lbryio/lbry-desktop#4652]
**Is your feature request related to a problem? Please describe.**
i just lost thousands of LBC (about 5k i think) and access to my channel(s)...

**Describe the solution you'd like**
expose the seed and allow recovering all important account+wallet info (including lbry.tv creds and anything else that gets added in the future) from that seed (like most crypto currencies do). 

The only thing that shouldn't be recovered is the downloaded files. Those can just be re-downloaded by the client the next time they are viewed. 

I see lbry already uses a seed in some way, but it is not exposed to the user in the GUI. i don't know why, but backing up directories (that may change names or locations over time) is not as convenient or user friendly as just backing up a seed. Also, lbry being the outlier, will make people forget that it doesn't use a seed. 

That's exactly what happened to me. I have all my seeds from other crypto currencies saved, so i thought i was good, and didn't double check when i was busy changing hardware and distros recently. So, everything was easy to recover/migrate except lbry, which let me down a lot, since i've been very bullish on lbry and was already using it for a channel.

thanks

**Comment by tzarebczan** [comment]
Sorry to hear about the trouble with losing the funds and channel. Shoot us an email at hello@lbry.com with the account details and we'll see what we can do. 

On the Desktop app, there's an option to backup the wallet on the Help page. If you don't have sync/backup enabled with us, on the wallet page it will tell you this and lead to that backup option.

LBRY is a bit different from other cryptos due to how we store channel data and how we sync wallets across multiple devices - channels have special keys created inside the wallet when the channel is created - so a seed is not enough to recover it, and due to cross device scenarios, there may be multiple accounts with seeds in a single wallet.

**Comment by ITwrx** [comment]
thanks for your response. i lost internet for nearly two weeks when you sent this and then i keep forgetting about it. :) 
I had already sent an email to hello@lbry.com, but they told me without the wallet file there was nothing they could do. Understandable, but if i had the wallet file i wouldn't need any help. :) Even if i i couldn't recover my LBC, regaining access to my channels would be nice. I'll send another email with reference to this github issue so the person responding knows what the situation is.

-----

TLDR; If at all possible, please consider redesign of the way this stuff works to transparently use a master seed for all accounts and devices, over current options of files backup or remote, centralized, server sync.

-----

I saw the sync option, but i don't like that option. Backing up to a remote server, especially one that is not in my control is not acceptable to me for a crypto wallet/account backup mechanism (i don't use a remote party sync for anything. I'm not happy with Firefox for the same reason. They have known people need to import and export passwords, without addons, for years, but they ignore the users and build their sync "solution". It's infuriating.). I also think it makes LBRY look bad to security/privacy conscious people to go this route. I understand that one can theoretically implement their own sync server, but that is still a security concern, if not as much of a privacy concern. I would only be satisfied with a local option, but not being responsible for raw files and folders. Those change too much, and are too easy to screw up. 

> LBRY is a bit different from other cryptos due to how we store channel data and how we sync wallets across multiple devices - channels have special keys created inside the wallet when the channel is created - so a seed is not enough to recover it, and due to cross device scenarios, there may be multiple accounts with seeds in a single wallet.

IMO, lbry needs a master [insert newspeak term here] seed for the user to use, regardless of how it needs to work in the background. Seeds make backing up, moving, and restoring super easy without compromising security and privacy.

This is why i opened a feature request. I consider the way these things work now as a design flaw that is affecting LBRY uptake and should be redesigned and reimplemented sooner rather than later, for devs' sake, if not users'. Though, admittedly, i'm ignorant on most/all of the technical details, and am just speaking from a high-level user perspective, regarding the disconnect between lbry-desktop and lbry.tv, (and possibly between different devices. i don't remember how that works) and the problems i've mentioned with the current backup methods, etc. 

It's just not smooth enough while remaining secure and private from the end user's naive perspective, imho. Do lbry devs not know how to re/design it to use a master seed, is it technically impossible/too difficult/time consuming, or LBRY devs just think the current solution are easier/good enough? I wonder if the community was asked their opinion prior to the sync and other seed related design decisions being made. I don't always ask about everything either. It depends on various things including the funding model. I'm not judging, just wondering. If that kind of openness works with LBRY's situation, then it might be in LBRY's interest to get prelim feedback in the future.  Maybe users have good design ideas, or have hard requirements devs might not realize. 

Maybe look at the LBRY documentation on these subjects with fresh, unbiased eyes to see what the user sees. Maybe you'll agree with me then. Maybe LBRY could create a simple poll for LBRY users to answer. The current backup/account sync methods, vs a master seed that is good for tv, desktop and any device. Thanks for your consideration either way. I would much rather LBRY be successful regardless.

**Comment by ITwrx** [comment]
update: somehow my LBRY wallet (once updated again) has my LBC and channels! Thanks a lot if someone helped with that!

**Comment by chaserene** [comment]
> channels have special keys created inside the wallet when the channel is created - so a seed is not enough to recover it

why don't you make the generation of those keys deterministic?

also, this issue is about the lack of a feature, and remains unresolved. please re-open it.
